src/state_machine.zig

```diff
@@ -437,19 +444,19 @@ pub fn StateMachineType(comptime Storage: type) type {
                                 count += 1;
                             }
                         } else {
-                            assert(result == .linked_event_failed);
+                            assert(result == .linked_event_failed or result == .linked_event_chain_open);
                         }
                     }
                     results[count] = .{ .index = @intCast(u32, index), .result = result };
                     count += 1;
                 }
-                if (!event.flags.linked and chain != null) {
+                if ((!event.flags.linked and chain != null) or
+                    result == .linked_event_chain_open)
+                {
```

Comment:
I had to add an `or` here to keep the assertion `assert(chain == null)` at the end.
Not sure about that.